scripts/alert_emails/create_lists_to_email.ts

```diff
@@ -1,59 +1,94 @@
 
 /**
- * Takes a list of fips to alert on and a list of users who want to get alert
- * and generates a set of users to email in firestore like:
- *     - snapshots/<id>/locations/<fips>/emails/<email> => {sent: null }
+ * Takes the alerts file generated by `generate_daily_alerts.ts` and updates the
+ * list of emails that we need to send for each location. The steps are:
  *
- * Takes in a file with {[fips: string]: {...data}} and generates documents
- * for each user to be emailed.
+ * 1. Read the locations that changed threat level from the alerts file.
+ * 2. Fetch the emails that are subscribed to the locations that changed threat level.
+ * 3. Creates a list of emails for each location and stores them in Firestore.
  *
- * You can run via `yarn create-lists-to-email fipsToAlertFilename currentSnapshot`
+ * Run the script via yarn:
+ *
+ *   $ yarn create-lists-to-email alerts.json snapshotId
  */
-(async () => {
-    const outputFolder = path.join(__dirname);
-    await fs.ensureDir(outputFolder);
-
-    const {fipsToAlertFilename, currentSnapshot} = await parseArgs();
-    const alertPath = path.join(outputFolder, fipsToAlertFilename);
-
-    const rawdata = fs.readFileSync(alertPath, "utf8");
-    const locationsWithAlerts: { [fips: string]: Alert } = JSON.parse(rawdata);
-
-    const db = getFirestore();
-    Object.keys(locationsWithAlerts).forEach(async (fips: string) => {
-        db.collection('alerts-subscriptions').where("locations", "array-contains", fips).get()
-            .then( (querySnapshot) => {
-                querySnapshot.forEach((doc) => {
-                    db.collection('snapshots').doc(`${currentSnapshot}/locations/${fips}/emails/${doc.id}`).set({
-                        sentAt:null,
-                    });
-                });
-                return;
-            }).catch((err: any) => {console.log(err)});
-    });
-
-    console.log(`Done. Wrote users to firestore`);
-})();
-
-async function parseArgs(): Promise<{fipsToAlertFilename: string, currentSnapshot: string}> {
-    const args = process.argv.slice(2);
-     if (args.length == 2) {
-        const fipsToAlertFilename = args[0];
-        const currentSnapshot = args[1];
-        return {fipsToAlertFilename, currentSnapshot};
-    } else {
-        exitWithUsage();
+async function updateSnapshotEmails(
+  alertsFileName: string,
+  snapshotId: string,
+) {
+  const db = getFirestore();
+  const alertsByLocation = readAlertsFile(alertsFileName);
+  const fipsList = Object.keys(alertsByLocation);
+
+  for (const fips in fipsList) {
+    const emailQuerySnapshot = await db
+      .collection('alerts-subscriptions')
+      .where('locations', 'array-contains', fips)
+      .get();
+
+    // Remove invalid and repeated emails for the location from the list
+    const emailList = _.uniq(
+      emailQuerySnapshot.docs
+        .map(emailDoc => emailDoc.id)
+        .filter(isValidEmail)
+        .map(_.toLower),
+    );
```

Comment:
Since we are going to use the snapshots to do analytics on the email subscriptions, I'm going to keep the cleaning for now. Later, we can write a script to do a one-time cleanup of the `alert-subscriptions` collection to remove all the invalid emails.